pennylane_lightning/src/Gates.cpp

```diff
@@ -682,6 +682,48 @@ void Pennylane::CGeneralRotationGate::applyKernel(
 
 // -------------------------------------------------------------------------------------------------------------
 
+const string Pennylane::CPhaseShiftGate::label = "ControlledPhaseShift";
+
+Pennylane::CPhaseShiftGate
+Pennylane::CPhaseShiftGate::create(const vector<double> &parameters) {
+    validateLength(Pennylane::CPhaseShiftGate::label, parameters, 1);
+    return Pennylane::CPhaseShiftGate(parameters[0]);
+}
+
+Pennylane::CPhaseShiftGate::CPhaseShiftGate(double phi)
+    : shift(std::pow(M_E, CplxType(0, phi))), matrix{1, 0, 0, 0,    0, 1,
+                                                     0, 0, 0, 0,    1, 0,
+                                                     0, 0, 0, shift} {}
```

Comment:
Maybe nice to have the matrix part shown in a 4x4 grid to make it visually easy to see what's going on.

Reply:
(unless clang requires it to be like the above)

Reply:
This is re-arranged by clang-format, indeed!

Reply:
Yep, I think we may need to revisit clang formatting rules at a later date to better fix these.